app/platform/fabric/service/UserService.js

```diff
@@ -142,6 +142,77 @@ class UserService {
 		};
 	}
 
+	/**
+	 *
+	 *
+	 * @param {*} user
+	 * @returns
+	 * @memberof UserService
+	 */
+	async unregister(user) {
+		try {
+			if (!user.user || !user.network) {
+				throw new Error('Invalid parameters');
+			}
+			const combinedUserName = `${user.network}-${user.user}`;
+
+			await Model.User.findOne({
+				where: {
+					username: combinedUserName
+				}
+			}).then(async userEntry => {
+				if (userEntry == null) {
+					throw new Error(`${user.user} does not exists`);
+				}
+
+				const newUser = {
+					username: combinedUserName,
+					truncate: true
+				};
+
+				await Model.User.destroy(newUser)
+					.then(() => {
+						return true;
+					})
+					.catch(error => {
+						throw new Error(`Failed to unRegister user : ${user.user} : ${error}`);
+					});
+			});
+		} catch (error) {
+			return {
+				status: 400,
+				message: error.toString()
+			};
+		}
+		return {
+			status: 200,
+			message: 'Unregistered successfully!'
+		};
+	}
+
+	async userlist() {
+		var userList = [];
+		try {
+			await Model.User.findAll()
+				.then(users => {
+					userList = users;
+					return true;
+				})
+				.catch(error => {
+					throw new Error(`Failed to get all users : ${error}`);
+				});
+		} catch (error) {
+			return {
+				status: 400,
+				message: error.toString()
+			};
+		}
+		return {
+			status: 200,
+			message: userList
```

Comment:
I don't think we need to return list of entire user information including hashed password. I think the following information is enough for user infromation which is returned.
- Username
- Email
- NetworkName
- FirstName
- LastName
- Roles

Reply:
Got it. Now returned only Suggested fields information . Please verify now